noxfile.py

```diff
@@ -107,6 +110,13 @@ def tests(session: PowerSession, coverage, pkg_specs):
     # Important: do not surround the command into double quotes as in the shell !
     # session.run('python', '-c', 'import os; os.chdir(\'./docs/\'); import %s' % pkg_name)
 
+    # Type checking is supported from python 3.7
+    if float(session.python) >= 3.7:
+        try:
+            session.run2("npm install -g pyright@1.1.247")
+        except CommandFailed:
+            print("Failed to install pyright, typing tests would be skipped")
```

Comment:
And please note this.

What do you think? We should fail here if pyright install failed?

Typing tests are run only if pyright is found, so it will safely pass tests when node is not installed for example.

But, it depends on importance of those tests. I think we could remove `try/except` here, but decision is yours to make.

Reply:
See first comment above: my opinion is that we should not force future contributors to have to install something else than miniconda to get started on this project. Therefore if we find a way to make everything installed by nox/conda/pip without npm, then I'll be aligned with you (mandatory install).

However if npm remains necessary, then I'll be in favour of keeping the try/catch, so that developers with only conda and not npm can still contribute without hassle. Anyway the GHA workflow will run these.